File: app/src/main/java/com/example/turnero/SignInFragmentAdapter.java
package com.example.turnero;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.lifecycle.Lifecycle;
import androidx.viewpager2.adapter.FragmentStateAdapter;

public class SignInFragmentAdapter extends FragmentStateAdapter {
    public SignInFragmentAdapter(FragmentManager fragmentManager, @NonNull Lifecycle lifecycle) {
        super(fragmentManager, lifecycle);
    }

    @NonNull
    @Override
    public Fragment createFragment(int position) {
//        creamos un switch para pasar entre tabs
        if (position == 1) {
            return new TabRegisterFragment();
        }


        return new TabLoginFragment();
    }

    @Override
    public int getItemCount() {
        return 2;
    }
}

File: app/src/main/java/com/example/turnero/adapters/SignInFragmentAdapter.kt
package com.example.turnero.adapters

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.lifecycle.Lifecycle
import androidx.viewpager2.adapter.FragmentStateAdapter
import com.example.turnero.TabRegisterFragment
import com.example.turnero.TabLoginFragment

class SignInFragmentAdapter(fragmentManager: FragmentManager?, lifecycle: Lifecycle) :
    FragmentStateAdapter(
        fragmentManager!!, lifecycle
    ) {
    override fun createFragment(position: Int): Fragment {
//        creamos un switch para pasar entre tabs
        return if (position == 1) {
            TabRegisterFragment()
        } else TabLoginFragment()
    }

    override fun getItemCount(): Int {
        return 2
    }
}